packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -73,6 +73,29 @@ export 'package:flutter/services.dart' show Brightness;
 // Examples can assume:
 // late BuildContext context;
 
+/// Defines custom adaptations to a [ThemeData] object.
+///
+/// This is typically used for adaptive themes, but this can be used to adapt any
```

Comment:
Although this type could be applied to use cases beyond "components with an `adaptive` factory constructor", I don't think it's useful to make that claim here (yet).

The most important thing to say here is: Currently, only [Switch.adaptive] supports this class.